core/templates/dev/head/pages/exploration_player/TutorCardDirective.js

```diff
@@ -76,12 +78,67 @@ oppia.directive('tutorCard', [
             $scope.interactionInstructions = (
               ExplorationPlayerStateService.getInteractionInstructions(
                 $scope.activeCard.stateName));
+            $scope.isCurrentHintUsable = false;
+            $scope.promiseForHint = $timeout(function () {
+              $scope.isCurrentHintUsable = true;
+            }, WAIT_FOR_HINT_MSEC);
+
+            $scope.currentInteractionHints = oppiaPlayerService.getInteraction(
+              $scope.activeCard.stateName).hints;
           };
 
           $scope.arePreviousResponsesShown = false;
 
           $scope.waitingForOppiaFeedback = false;
 
+          $scope.isCurrentHintUsable = false;
+
+          $scope.allHintsExhausted = false;
+
+          $scope.promiseForHint = null;
+
+          $scope.hintButtonClicks = 0;
+
+          $scope.tooltipText = '';
+
+          $scope.showHint = function() {
+            var hints = $scope.currentInteractionHints;
+            if ($scope.hintButtonClicks < hints.length) {
+              var currentHint = hints[$scope.hintButtonClicks].hintText;
+              playerTranscriptService.addNewAnswer('I would like a hint.');
+              $timeout(function() {
+                $scope.waitingForOppiaFeedback = false;
+                playerTranscriptService.addNewFeedback(currentHint);
+              }, 600);
+              $scope.isCurrentHintUsable = false;
+              $scope.promiseForHint = $timeout(function() {
+                $scope.isCurrentHintUsable = true;
+              }, WAIT_FOR_HINT_MSEC);
+
+              if ($scope.hintButtonClicks === hints.length - 1) {
+                $scope.isCurrentHintUsable = false;
+                $scope.allHintsExhausted = true;
+                $scope.hintButtonClicks = 0;
+              } else {
+                $scope.hintButtonClicks += 1;
+              }
+            }
+          };
+
+          $scope.isHintAvailable = function() {
```

Comment:
General note: We should not be duplicating all this code. Consider pulling it out into a common service.